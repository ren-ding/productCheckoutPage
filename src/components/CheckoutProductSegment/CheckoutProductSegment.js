import React, {Component} from 'react';
import PropTypes from 'prop-types';

export default class CheckoutProductSegment extends Component {
    render() {
        return (
            <tr>
                <td>{this.props.product.productId}</td>
                <td>{this.props.product.quantity}</td>
                <td>
                    <div className="remove-product-from-checkout ui button"
                        onClick={()=> this.props.removeFromCheckout(this.props.product.productId)}
                    >
                        <i className="close icon"></i>
                    </div>
                </td>
            </tr>
        );
    }
}

CheckoutProductSegment.propTypes = {
    product: PropTypes.object.isRequired,
    removeFromCheckout: PropTypes.func.isRequired
};