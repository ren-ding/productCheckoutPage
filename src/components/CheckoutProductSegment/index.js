import CheckoutProductSegment from './CheckoutProductSegment';
export default CheckoutProductSegment;