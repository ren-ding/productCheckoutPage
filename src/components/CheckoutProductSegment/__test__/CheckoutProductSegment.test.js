import React from 'react';
import CheckoutProductSegment from '../CheckoutProductSegment';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });

describe('CheckoutProductSegment', ()=> {
    let props;
    let checkoutProductSegment;

    beforeEach(()=> {
        props = {
            product: {},
            removeFromCheckout: ()=>{}
        }

        checkoutProductSegment = () => shallow(<CheckoutProductSegment {...props} />);    
    });

    describe('render', () => {
       it('should render to match snapshot', () => {
           const tree = renderer.create(<CheckoutProductSegment {...props}/>).toJSON();
           expect(tree).toMatchSnapshot();
       }); 
    });

    describe('click add button', ()=> {
        it('should call addToCheckout once with productid', ()=> {
            const mockRemoveFromCheckout = jest.fn();
            props = {
                product:{
                    productId: 'wf',
                    quantity:1
                },
                removeFromCheckout: mockRemoveFromCheckout
            }
            const segment = checkoutProductSegment();
            
            segment.find('.remove-product-from-checkout').simulate('click');
            expect(mockRemoveFromCheckout).toHaveBeenCalled();
            expect(mockRemoveFromCheckout.mock.calls.length).toBe(1);
            //first call of the first argument is the productId
            expect(mockRemoveFromCheckout.mock.calls[0][0]).toBe('wf');
        });
    });
});