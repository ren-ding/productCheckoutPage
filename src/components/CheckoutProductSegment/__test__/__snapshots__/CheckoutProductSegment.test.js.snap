// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutProductSegment render should render to match snapshot 1`] = `
<tr>
  <td />
  <td />
  <td>
    <div
      className="remove-product-from-checkout ui button"
      onClick={[Function]}
    >
      <i
        className="close icon"
      />
    </div>
  </td>
</tr>
`;
